components/ArchiveList.php

```diff
@@ -0,0 +1,98 @@
+    {
+        return [
+            'name'        => 'rainlab.blog::lang.settings.archive_list_name',
+            'description' => 'rainlab.blog::lang.settings.archive_list_description'
+        ];
+    }
+
+    public function defineProperties()
+    {
+        return [
+            'monthsToShow' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_months_to_show',
+                'type'              => 'string',
+                'validationPattern' => '^[0-9]+$',
+                'validationMessage' => 'rainlab.blog::lang.settings.archive_list_months_to_show_validation',
+                'default'           => '5',
+            ],
+            'showPostCount' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_show_post_count',
+                'type'              => 'checkbox',
+                'default'           => 'false'
+            ],
+            'slug' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_slug',
+                'description'       => 'rainlab.blog::lang.settings.archive_list_slug_description',
+                'default'           => '{{ :slug }}',
+                'type'              => 'string'
+            ],
+            'archivePage' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_page',
+                'description'       => 'rainlab.blog::lang.settings.archive_list_page_description',
+                'type'              => 'dropdown',
+                'default'           => 'blog/archive',
+                'group'             => 'Links',
+            ],
+        ];
+    }
+
+    public function getArchivePageOptions()
+    {
+        return Page::sortBy('baseFileName')->lists('baseFileName', 'baseFileName');
+    }
+
+    public function onRun()
+    {
+        $this->archiveMonths = $this->page['archiveMonths'] = $this->loadMonths($this->property('monthsToShow'));
+        $this->currentArchiveSlug = $this->page['currentArchiveSlug'] = $this->property('slug');
+        $this->archivePage = $this->page['archivePage'] = $this->property('archivePage');
+    }
+
+    protected function loadMonths($monthsToShow)
+    {
+        if (($monthsToShow > 0) ? $limit = $monthsToShow : $limit = 100);
+
+        $archiveRange = BlogPost::select(DB::raw("DATE_FORMAT(published_at, '%M %Y') AS month, count(*) as count"))
+                                ->where('published', 1)
+                                ->orderBy('published_at', 'ASC')
+                                ->limit($limit)
+                                ->groupBy(DB::raw("DATE_FORMAT(published_at, '%M %Y')"))
+                                ->get();
+
+        foreach ($archiveRange as $post) {
+            $post->url = $this->controller->pageUrl($this->property('archivePage'), ['slug' => urlencode($post->month)]);
+            
+            if (!$this->property('showPostCount')) {
```

Comment:
Why is this required? If you want to toggle the display of the count of posts in a month, then shouldn't that be done in the rendering, not the data structure?